corelibs/libcache.md

```diff
@@ -0,0 +1,187 @@
+# Cache library
+
+`libcache` is a thread-safe library which implements a n-way set-associative cache. It provides the user with a quicker access to data upon prior mapping from a cached source memory on to the cache table.
+<!-- TODO: mention whether the library is static and precompiled -->
+
+![image](_images/libcache.png)
+
+## Contents
+- [API](#API)
+    - [Data types](#data-types)
+    - [Functions](#functions)
+- [Configurable cache parameters](#Configurable-cache-parameters)
+- [Implementation](#implementation)
+    - [Overview](#overview)
+    - [Organization](#organization)
+    - [Bitmasks](#bitmasks)
+    - [Data structures](#data-structures)
+- [Cache operations](#cache-operations)
+    - [Read](#reading-from-the-cache-to-a-buffer)
+    - [Write](#writing-a-buffer-to-the-cache)
+    - [Flush](#flushing-the-cache)
+    - [Invalidate](#invalidating-the-cache)
+- [Running tests](#running-tests)
+- [Navigation links](#See-also)
+
+## API
+The `libcache` library is a collection of types and functions which provides the user with tools necessary to create, operate and destroy a cache.
+<br /><br />
+The user is obligated to supply suitable, source memory-specific callbacks that execute write to and read from the cached source memory as well as a definition of a device driver context.
+
+### Data types
+| Type | Description | Remarks |
+| ---- | ------------| ------- |
+`cachectx_t` | Cache context — represents the cache table | Opaque type — can only be accessed and/or modified through/by provided API functions.
+`cache_devCtx_t` | Device driver context | **A cached source memory-specific type definition ought to be supplied by the user.** <br /><br /> |
```

Comment:
I think user has to provide `struct cache_devCtx`, not `cache_devCtx_t`

Reply:
Also, it's part of `cache_ops_t`, as well as callbacks (that are not described here)